### DISCUSSION TITLE
Unsupported Type "LatLng"

### DISCUSSION THREAD

**Discussion by weberscode** [isar/isar#757] [Help]
Hy everybody,

i got an "Unsupported type" error for variable of type LatLng.

This is my class:

import 'package:isar/isar.dart';
import 'package:latlong2/latlong.dart';

part 'location.g.dart';

@Collection()
class Location {
  LatLng? latlng;
}

This is the error message:

![grafik](https://user-images.githubusercontent.com/72982682/194758960-f24fb7db-7af6-46a1-93b8-98dceee499b4.png)

What is the problem? How to solve it?

Thanks for all answers.

Simon

**Comment by MrCsabaToth** [comment]
I guess LatLng is not one of the supported basic types. Try to handle it as an embedded object?

**Comment by TimeLord2010** [comment]
According to the documentation, you need to anottate classes to that to work:

```dart
@collection
class Email {
  Id? id;

  String? title;

  Recepient? recipient;
}

@embedded
class Recepient {
  String? name;

  String? address;
}
```

https://isar.dev/schema.html#embedded-objects
